plugins/logitech-rallysystem/fu-logitech-rallysystem-plugin.c

```diff
@@ -21,6 +23,51 @@ fu_logitech_rallysystem_plugin_init(FuLogitechRallysystemPlugin *self)
 {
 }
 
+static void
+fu_logitech_rallysystem_plugin_finalize(GObject *object)
+{
+	FuLogitechRallysystemPlugin *self = FU_LOGITECH_RALLYSYSTEM_PLUGIN(object);
+	g_free(self->system_version);
+	G_OBJECT_CLASS(fu_logitech_rallysystem_plugin_parent_class)->finalize(object);
+}
+
+static void
+fu_logitech_rallysystem_plugin_device_added(FuPlugin *plugin, FuDevice *device)
+{
+	FuLogitechRallysystemPlugin *self = FU_LOGITECH_RALLYSYSTEM_PLUGIN(plugin);
+	GPtrArray *devices;
+
+	/*
+	 * composite device is composed of multiple sub-devices: audio, video, tablehub, speakers.
+	 * Each sub-device has their own unique firmware version.
+	 * Only audio sub-device has topology information of all sub-devices.
+	 * Only audio sub-device has information of overall/unique system version.
+	 * Only tablehub device is responsible to push firmware images to all sub-devices,
+	 * all sub-devices update firmware independently, and provide status to tablehub.
+	 * Since only tablehub can accept firmware image, its guid is used in .metainfo.xml file.
+	 * Ask here is for tablehub, to provide system version, so that application can use single
+	 * guid to query system version and check .metainfo.xml to determine if upgraded is needed
+	 * or not. Following logic preserve the system version information from audio sub-device and
+	 * overwrite local version information of tablehub with this system version
+	 */
+	if (g_strcmp0(fu_device_get_plugin(device), "logitech_rallysystem") == 0) {
+		if (FU_IS_LOGITECH_RALLYSYSTEM_AUDIO_DEVICE(device)) {
+			self->system_parent_plugin = plugin;
```

Comment:
I don't think you should store the pointer -- can you instead check the `fu_device_get_plugin()` string is `logitech_rallysystem` instead?